types/react-redux/index.d.ts

```diff
@@ -49,13 +53,72 @@ export interface InferableComponentDecorator<TOwnProps> {
  */
 export declare function connect<TOwnProps>(): InferableComponentDecorator<TOwnProps>;
 
+export declare function connect<TStateProps, undefined, TOwnProps>(
+    mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>
+): ComponentDecorator<TStateProps, void, TOwnProps>;
+
+export declare function connect<undefined, TDispatchProps, TOwnProps>(
+    mapStateToProps: null | undefined,
+    mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>
+): ComponentDecorator<void, TDispatchProps, TOwnProps>;
+
 export declare function connect<TStateProps, TDispatchProps, TOwnProps>(
-    mapStateToProps?: MapStateToProps<TStateProps, TOwnProps> | MapStateToPropsFactory<TStateProps, TOwnProps>,
-    mapDispatchToProps?: MapDispatchToProps<TDispatchProps, TOwnProps> | MapDispatchToPropsFactory<TDispatchProps, TOwnProps>,
-    mergeProps?: MergeProps<TStateProps, TDispatchProps, TOwnProps>,
-    options?: Options
+    mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>,
+    mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>
 ): ComponentDecorator<TStateProps, TDispatchProps, TOwnProps>;
 
+export declare function connect<TStateProps, undefined, TOwnProps, TMergedProps>(
```

Comment:
Did not know you could use `undefined` like this, but neat. I _think_ this is desirable to keep the positional values lined up across all the overloads, but am open to being swayed.